### PR TITLE
Flight data migrations for Visual rules

### PR DESCRIPTION
**Pull request by pasanchMSFT**
### Summary of changes 🔍 
PR to conditionally apply migration to config data in case flighting for Visual rules is turned on.

### Testing 🧪
N/A

### Checklist ✔️
- [x] Linked associated issue (if present)
- [x] Added relevant labels
- [x] Requested developer reviews
- [x] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

File: src/Adapters/BlobAdapter.ts [hunk 1]
```diff
@@ -135,8 +140,10 @@ export default class BlobAdapter implements IBlobAdapter {
 
             try {
                 const configBlob = await getConfigBlob();
-                // Handle any schema migrations to transform legacy data into valid data
-                handleMigrations(configBlob.data);
+                if (showVisualRulesFeature) {
+                    // Handle any schema migrations to transform legacy data into valid data
+                    handleMigrations(configBlob.data);
```
Comment (iremgokce) [inline src/Adapters/BlobAdapter.ts:145]: do we need to do null check on data before passing to handleMigration?

File: src/Adapters/BlobAdapter.ts [hunk 2]
```diff
@@ -28,6 +29,10 @@ import defaultConfig from './__mockData__/3DScenesConfiguration.default.json';
 import { ComponentError } from '../Models/Classes';
 import { handleMigrations, LogConfigFileTelemetry } from './BlobAdapterUtility';
 
+const showVisualRulesFeature =
+    localStorage.getItem(
+        LOCAL_STORAGE_KEYS.FeatureFlags.VisualRules.showVisualRulesFeature
+    ) === 'true' || false;
```
Comment (iremgokce) [inline src/Adapters/BlobAdapter.ts:35]: why not just this?
```suggestion
    ) === 'true';
```

**Review by iremgokce — APPROVED** [review]
put two comments

File: src/Adapters/BlobAdapter.ts (hunk 1 above)
Comment (pasanchMSFT) [reply src/Adapters/BlobAdapter.ts:145]: `handleMigration` methods have null checks in them

**Review by msnyder-msft — APPROVED** [review]

**Pull request merged**